src/main/java/org/opensearch/neuralsearch/rest/RestNeuralStatsHandler.java

```diff
@@ -0,0 +1,157 @@
+import org.opensearch.neuralsearch.settings.NeuralSearchSettingsAccessor;
+import org.opensearch.neuralsearch.stats.NeuralStatsInput;
+import org.opensearch.neuralsearch.stats.events.EventStatName;
+import org.opensearch.neuralsearch.stats.state.StateStatName;
+import org.opensearch.neuralsearch.transport.NeuralStatsAction;
+import org.opensearch.neuralsearch.transport.NeuralStatsRequest;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestActions;
+import org.opensearch.transport.client.node.NodeClient;
+
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.opensearch.neuralsearch.plugin.NeuralSearch.NEURAL_BASE_URI;
+import static org.opensearch.neuralsearch.processor.util.RestActionUtils.splitCommaSeparatedParam;
+
+@Log4j2
+@AllArgsConstructor
+public class RestNeuralStatsHandler extends BaseRestHandler {
+    private static final String NAME = "neural_stats_action";
+    public static final String FLATTEN_PARAM = "flat_keys";
+    public static final String INCLUDE_METADATA_PARAM = "include_metadata";
+
+    private static final Set<String> EVENT_STAT_NAMES = EnumSet.allOf(EventStatName.class)
+        .stream()
+        .map(EventStatName::getNameString)
+        .map(String::toLowerCase)
+        .collect(Collectors.toSet());
+
+    private static final Set<String> STATE_STAT_NAMES = EnumSet.allOf(StateStatName.class)
+        .stream()
+        .map(StateStatName::getNameString)
+        .map(String::toLowerCase)
+        .collect(Collectors.toSet());
+
+    private NeuralSearchSettingsAccessor settingsAccessor;
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ImmutableList.of(
+            new Route(RestRequest.Method.GET, NEURAL_BASE_URI + "/{nodeId}/stats/"),
+            new Route(RestRequest.Method.GET, NEURAL_BASE_URI + "/{nodeId}/stats/{stat}"),
+            new Route(RestRequest.Method.GET, NEURAL_BASE_URI + "/stats/"),
+            new Route(RestRequest.Method.GET, NEURAL_BASE_URI + "/stats/{stat}")
+        );
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
+        if (settingsAccessor.getIsStatsEnabled() == false) {
+            // Process params, or else will automatically return a 400 instead of a 403
+            splitCommaSeparatedParam(request, "nodeId");
+            splitCommaSeparatedParam(request, "stat");
+            request.paramAsBoolean(FLATTEN_PARAM, false);
+            request.paramAsBoolean(INCLUDE_METADATA_PARAM, false);
+
+            return channel -> channel.sendResponse(new BytesRestResponse(RestStatus.FORBIDDEN, "Stats endpoint is disabled"));
+        }
```

Comment:
Move this to a separate dedicated method as validation method.

QQ: why do you need to call split methods and updating request if you anyway sending forbidden request exception at line 80.